### PR TITLE
feat(collection): Add removable support

### PR DESCRIPTION
**Pull request by NicholasBoll**
## Summary

Fixes: #3025

Adds `remove` event to the `ListModel`. An `onRemove` config should be added to dynamic lists to remove the item from the collection.

The `MultiSelect.Input` uses this new remove event to handle removing items from the Selected pill list when the user uses the “Delete” key. Focus is managed by the collection system when an item is removed.

## Release Category
Components

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [x] Label `ready for review` has been added to PR

## For the Reviewer

- [x] PR title is short and descriptive
- [x] PR summary describes the change (Fixes/Resolves linked correctly)

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation
- [ ] Testing

### CODE REVIEW

File: modules/preview-react/multi-select/lib/MultiSelectInput.tsx [hunk 1]
```diff
@@ -228,34 +186,25 @@ export const MultiSelectSearchInput = createSubcomponent(TextInput)({
...
+            {...formInputProps}
+          />
           <InputGroup.Input
             data-part="user-input"
             as={Element}
             aria-labelledby={ariaLabelledBy}
             {...elemProps}
           />
-          <InputGroup.InnerEnd>
+          <InputGroup.InnerEnd width={system.space.x4}>
```
Comment (mannycarrera4) [inline modules/preview-react/multi-select/lib/MultiSelectInput.tsx:200]: do we care if this is a style prop or should we use `cs` here?

**Review by josh-bagwell — APPROVED** [review]

**Comment by josh-bagwell** [comment]
Seeing two failed Tabs specs

File: modules/preview-react/multi-select/lib/MultiSelectInput.tsx (hunk 1 above)
Comment (NicholasBoll) [reply modules/preview-react/multi-select/lib/MultiSelectInput.tsx:200]: `width` is a CSS variable in inputgroup, so `cs={{width...}}` would now work correctly.

**Review by mannycarrera4 — APPROVED** [review]

File: cypress/component/Tabs.spec.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ describe('Tabs', () => {
 
         context('when the tab key is pressed', () => {
           beforeEach(() => {
-            cy.tab();
+            cy.realPress('Tab');
```
Comment (josh-bagwell) [inline cypress/component/Tabs.spec.tsx:78]: realPress or realType?

Comment (NicholasBoll) [reply cypress/component/Tabs.spec.tsx:78]: There's no `realType` for the tab key. There's only a `realPress`. It probably makes more sense to use `realPress` for keyboard commands and `readType` or just `type` for entering in text.

**Pull request merged**